src/chase.rs

```diff
@@ -191,11 +191,10 @@ pub fn solve_all<'s, S, M, Sel, Stg, E, B>(strategy: &mut Stg, evaluator: &E, bo
           E: EvaluatorTrait<'s, Sel, B, Sequent=S, Model=M>,
           B: BounderTrait {
     let mut result: Vec<M> = Vec::new();
-    let mut f = |m: M| result.push(m);
     while !strategy.empty() {
-        solve(strategy, evaluator, bounder, &mut f);
+        solve(strategy, evaluator, bounder, |m| result.push(m));
```

Comment:
oh I guess I missed this one. But wondering why the compiler didn't show a warning?

Reply:
Because a reference to a closure still implements the FnMut trait.

Reply:
ah I see. my bad! I forgot that solve now takes the closure, not a reference